File: Dev/C++/Mascaret/src/VEHA/Entity/PointSpecification.cpp
#include "MascaretApplication.h"
#include "VEHA/Entity/PointSpecification.h"
#include "VEHA/Kernel/PrimitiveType.h"
#include "VEHA/Kernel/Model.h"
#include "Tools/utils.h"
#include "VEHA/Entity/VirtualRealityComponentFactory.h"

namespace VEHA
{

PointSpecification::PointSpecification()
:TopologicalSpecification(MascaretApplication::getInstance()->getModel()->getBasicType("point"))
{
}

PointSpecification::~PointSpecification()
{
}
shared_ptr<ValueSpecification> PointSpecification::clone() const
{
	//throw std::runtime_error("can't clone a PointSpecification");
	shared_ptr<PointSpecification> p=MascaretApplication::getInstance()->getVRComponentFactory()->createPoint(getGlobalPosition(),getGlobalRotation());

	return p;
}

}

File: Dev/CS/Mascaret/Mascaret/VEHA/Entity/PointSpecification.cs
using System;
using System.Collections.Generic;

namespace Mascaret
{
    public abstract class PointSpecification : TopologicalSpecification
    {

        private PointSpecification parent = null;
        public PointSpecification Parent
        {
            get { return parent; }
            set { parent = value; }
        }

        public PointSpecification()
            : base(MascaretApplication.Instance.Model.getBasicType("point"))
        {
        }

        public override ValueSpecification clone()
        {
            //	PointSpecification p = new UnityPointSpecification();//getGlobalPosition(),getGlobalRotation());
            //	return p;

            return null;
        }

        public abstract Vector3 getLocalPosition();
        public abstract void setLocalPosition(Vector3 pos);
        public abstract Vector3 getGlobalPosition();
        public abstract void setGlobalPosition(Vector3 pos);

        public abstract void translate(Vector3 pos);

        public abstract RotationVector getLocalRotation();
        public abstract void setLocalRotation(RotationVector orientation);
        public abstract RotationVector getGlobalRotation();
        public abstract void setGlobalRotation(RotationVector orientation);
        public abstract void rotate(RotationVector orientation);
        public abstract void rotate(Vector3 orientation);

        public abstract void setLocalOrientation(Vector3 ori);
        public abstract Vector3 getLocalOrientation();
        public abstract void setGlobalOrientation(Vector3 ori);
        public abstract Vector3 getGlobalOrientation();

        public abstract void globalToLocalLocation(PointSpecification pointInOut);
        public abstract void localToGlobalLocation(PointSpecification pointInOut);

        public abstract void globalToLocalPosition(Vector3 posInOut);
        public abstract void localToGlobalPosition(Vector3 posInOut);

        public abstract void localToGlobalOrientation(Vector3 oriInOut);
        public abstract void globalToLocalOrientation(Vector3 oriInOut);
    }
}

